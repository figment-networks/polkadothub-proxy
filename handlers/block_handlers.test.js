const blockHandlers = require('./block_handlers');
const {createApi} = require('../tests/shared');

// These actually DO hit the API
describe('integration', () => {
  let call = {request: {height: 2003930}};
  let api;

  beforeAll(async (done) => {
    api = await createApi();
    done()
  });

  describe('getByHeight', () => {
    it('returns a result that matches snapshot', async () => {
      expect(await blockHandlers.getByHeight(api, call)).toMatchSnapshot();
    });
  });

});
