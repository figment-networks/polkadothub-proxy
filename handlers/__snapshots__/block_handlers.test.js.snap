// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration getByHeight returns a result that matches snapshot 1`] = `
Object {
  "block": Object {
    "extrinsics": Array [
      Object {
        "args": "1587626346000",
        "extrinsicIndex": 0,
        "hash": "0x7fcaf3ffcaabfb6ab0d98df57f523381a3aef897da243a1d2ebe55df3e2e9924",
        "isSignedTransaction": false,
        "isSuccess": true,
        "method": "set",
        "nonce": 0,
        "section": "timestamp",
        "signature": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
        "signer": "CaKWz5omakTK7ovp4m3koXrHyHb7NG3Nt7GENHbviByZpKp",
      },
      Object {
        "args": "2003927",
        "extrinsicIndex": 1,
        "hash": "0xb5f7ebc1d0c875218330a395190cf943a089898950636a8f0fc80afa8b5453ee",
        "isSignedTransaction": false,
        "isSuccess": true,
        "method": "finalHint",
        "nonce": 0,
        "section": "finalityTracker",
        "signature": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
        "signer": "CaKWz5omakTK7ovp4m3koXrHyHb7NG3Nt7GENHbviByZpKp",
      },
      Object {
        "args": "[]",
        "extrinsicIndex": 2,
        "hash": "0xcf52705d1ade64fc0b05859ac28358c0770a217dd76b75e586ae848c56ae810d",
        "isSignedTransaction": false,
        "isSuccess": true,
        "method": "setHeads",
        "nonce": 0,
        "section": "parachains",
        "signature": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
        "signer": "CaKWz5omakTK7ovp4m3koXrHyHb7NG3Nt7GENHbviByZpKp",
      },
      Object {
        "args": "[{\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":671,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":672,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":673,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":674,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":675,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":676,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":677,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":678,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":679,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":680,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":681,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":682,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",1]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":683,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":684,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":685,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":686,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":687,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":688,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":689,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":690,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":691,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":692,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":693,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":694,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":695,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",1]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":696,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",6],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",5]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":697,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",5],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",4]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":698,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",6],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":699,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",6],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",7]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":700,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",6],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",7]]}}]",
        "extrinsicIndex": 3,
        "hash": "0x49cc0b81fce3d73dd30b125981ed511addbf121e8bdfed677c77da0d6162b944",
        "isSignedTransaction": true,
        "isSuccess": true,
        "method": "batch",
        "nonce": 42,
        "section": "utility",
        "signature": "0x54cb9be9b44a25b2493512ab9d413c0a6388f6bc4b607693b99d6e5230efc923669e7d4dd621d48dc67301b0380690f9d4994213e0e7d7b88cb3adcafdbe8582",
        "signer": "HA4FRTrk4n5f2gvsfNLYY4b9Tw5eincNBXpxpSV21qGVXvB",
      },
    ],
    "header": Object {
      "extrinsicsRoot": "0x19a41bde8a296637ee7c5b8cb52c9d0fdd44ec801d693e4b5a0d32e2fb6bbc0b",
      "height": 2003930,
      "parentHash": "0x9206b8381c247c728bbbb275f3c85129b180b91f6b3f96225ccbf6ee371ab251",
      "stateRoot": "0x6e0c7679036d9555dc77ffeb71d79fa66d5fc488c4316046c93a916f0079f1b4",
      "time": Object {
        "nanos": 0,
        "seconds": 1587626346,
      },
    },
  },
  "blockHash": "0x4d45265d2a20e0038cc1f5892502a699f09cc0123fc8303b2229eab27cb7e5c0",
  "chain": "Kusama",
  "era": 701,
  "session": 3517,
  "specVersion": "1055",
}
`;
